### PR TITLE
[ONNX] Use `VerificationOptions` to wrap option arguments

### PR DESCRIPTION
**Pull request by BowenBao**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #89947
* #89946
* #89808
* __->__ #89807

### CODE REVIEW

File: torch/onnx/verification.py [hunk 1]
```diff
@@ -30,6 +31,45 @@
...
+        remained_onnx_input_idx: If provided, only the specified inputs will be passed
+            to the ONNX model. Supply a list when there are unused inputs in the model.
+            Since unused inputs will be removed in the exported ONNX model, supplying
+            all inputs will cause an error on unexpected inputs. This parameter tells
+            the verifier which inputs to pass into the ONNX model.
+        acceptable_error_percentage: acceptable percentage of element mismatches in comparison.
+            It should be a float of value between 0.0 and 1.0.
+    """
+
+    flatten: bool = dataclasses.field(default=True)
```
Comment (justinchuby) [inline torch/onnx/verification.py:62]: Can simply be `= True`

**Review by justinchuby — APPROVED** [review]

**Review by titaiwangms — APPROVED** [review]
Nice

**Pull request closed without merging**